core/app/models/spree/shipment.rb

```diff
@@ -36,6 +36,7 @@ def generate_number(options = {})
     scope :ready,   -> { with_state('ready') }
     scope :shipped, -> { with_state('shipped') }
     scope :trackable, -> { where("tracking IS NOT NULL AND tracking != ''") }
+    scope :editable, -> { where(state: ["pending", "ready"]) }
```

Comment:
to me it doesn't seem that ready shipments should be editable. ready seems to convey that it is ready to ship such that a 3rd party should expect that it will not change and can safely ship this package.